zerver/actions/realm_settings.py

```diff
@@ -551,6 +560,13 @@ def do_deactivate_realm(
         event = dict(type="realm", op="deactivated", realm_id=realm.id)
         send_event_on_commit(realm, event, active_user_ids(realm.id))
 
+        if deletion_delay_days == 0:
+            event = {
+                "type": "scrub_deactivated_realm",
+                "realm_id": realm.id,
+            }
+            queue_json_publish("deferred_work", event)
```

Comment:
This was suggested by @alexmv in [this comment](https://github.com/zulip/zulip/pull/26109#discussion_r1242558033).